backend/src/chats/chats.gateway.ts

```diff
@@ -43,7 +59,38 @@ export class ChatsGateway {
       chatId: data.chatId,
     });
 
-    // Send the new message back to all clients
-    this.server.emit('newMessage', newMessage);
+    this.server.to(`chat:${data.chatId}`).emit('newMessage', newMessage);
+  }
+
+  @SubscribeMessage('joinChat')
+  async handleJoinChat(
+    @ConnectedSocket() client: Socket,
+    @User() user: UserEntity,
+    @MessageBody() chatId: number,
+  ) {
+    client.join(`chat:${chatId}`);
+    client.to(`chat:${chatId}`).emit('userJoined', user);
```

Comment:
👌👌

Reply:
Se tu der um `client.to` vai responder só pra quem deu o handleJoin, mas tu quer responder pra todos que tão naquele chat - acho que seria um `server.to`
```suggestion
    client.join(`chat:${chatId}`);
    this.server.to(`chat:${chatId}`).emit('userJoined', user);
```